openfeature/api.go

```diff
@@ -211,6 +235,46 @@ func (api *evaluationAPI) initNewAndShutdownOld(newProvider FeatureProvider, old
 	}(v)
 }
 
+// initNewAndShutdownOldSync is a helper to initialise new FeatureProvider and shutdown the old FeatureProvider.
+// Operations happen synchronously.
+func (api *evaluationAPI) initNewAndShutdownOldSync(newProvider FeatureProvider, oldProvider FeatureProvider) {
+	v, ok := newProvider.(StateHandler)
+	if ok && v.Status() == NotReadyState {
+		err := v.Init(api.evalCtx)
+		if err != nil {
+			api.eventExecutor.eventChan <- eventPayload{
+				Event{
+					ProviderName:         newProvider.Metadata().Name,
+					EventType:            ProviderError,
+					ProviderEventDetails: ProviderEventDetails{},
+				}, newProvider,
+			}
+		} else {
+			api.eventExecutor.eventChan <- eventPayload{
+				Event{
+					ProviderName:         newProvider.Metadata().Name,
+					EventType:            ProviderReady,
+					ProviderEventDetails: ProviderEventDetails{},
+				}, newProvider,
+			}
+		}
```

Comment:
This is duplicate of the same calls of https://github.com/open-feature/go-sdk/blob/main/openfeature/api.go#L174-L194 

Instead of duplication, you can extract this code to a dedicated function (ex:- `func initHelper(...)`) . Then you can call this function as a go routine for async operation (L174-L194 ) and as a direct function call for sync operation.

Reply:
Refactored and avoids duplication